lib/viewmodels/home_view_model.dart

```diff
@@ -1,18 +1,23 @@
 import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
 import "package:proxima/models/ui/post_overview.dart";
 import "package:proxima/services/database/post_repository_service.dart";
 import "package:proxima/services/database/user_repository_service.dart";
 import "package:proxima/services/geolocation_service.dart";
 
 /// This viewmodel is used to fetch the list of posts that are displayed in the home feed.
-/// It fetches the posts from the database and returns a list of [PostOverview] objects to be displayed.
-class HomeViewModel extends AsyncNotifier<List<PostOverview>> {
+/// It fetches the posts from the database and returns a list of
+/// (postId: [PostIdFirestore], postOverview: [PostOverview]) objects to be displayed.
+/// These represent the overview data to be displayed associated to the corresponding post id.
+class HomeViewModel extends AsyncNotifier<
+    List<({PostIdFirestore postId, PostOverview postOverview})>> {
```

Comment:
Also here this code should not make use of a tuple type as it hinders readably and usability of this type.
Why not simply put the `PostIdFirestore postId` inside the `PostOverview` type as a new field ?
I think it would make much more sense that way.

Reply:
This change would also simplify the map in the `HomeViewModel.build` method.

Reply:
My initial idea was that `PostOverview` represents only the data that are displayed on the screen.
Thus, adding a `PostIdFirestore` to it would break this abstraction, that's why I chose to return a list of tuples instead.
But effectively, if you think that this will contribute the code readibility, I can add the `postId` into the `PostOverview` directly.

Reply:
Yeah its not only for displayed elements, it's everything the view layer might need.